### PR TITLE
EN6 87 Content Settings Crop Ratios

### PR DESCRIPTION
**Pull request by raxkaynan**
Included:
* Crop Ratios section in the Content Settings page
* Ability to add crop ratio
* Ability to delete crop ratio
* Ability to edit crop ratio

### CODE REVIEW

File: src/ui/content-settings/ContentSettingsCropRatioInput.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    const { isNew } = this.props;
+    const { value } = this.state;
+
+    const renderedBtn = isNew ? (
+      <Button
+        data-test-id="content-settings-crop-ratio-input-add"
+        bsStyle="primary"
+        onClick={this.handleAddClick}
+      >
+        Add
```
Comment (jeffgo10) [inline src/ui/content-settings/ContentSettingsCropRatioInput.js]: use `<FormattedMessage>` for this.

File: src/state/content-settings/reducer.js [hunk 2]
```diff
@@ -1,6 +1,14 @@
-import { SET_CONTENT_SETTINGS, SET_EDITOR_SETTINGS } from 'state/content-settings/types';
+import {
+  SET_CONTENT_SETTINGS,
+  SET_EDITOR_SETTINGS,
+  SET_CROP_RATIOS,
+} from 'state/content-settings/types';
 
-const reducer = (state = {}, action = {}) => {
+const initialState = {
```
Comment (jeffgo10) [inline src/state/content-settings/reducer.js]: is it possible to only use `const initialState = {};` for this part?

Comment (raxkaynan) [reply src/state/content-settings/reducer.js]: The default `cropRatio`, which is set to an empty array, is used in connected components. Another approach would be to have default parameters or default props in a component that uses it. But, it's actually a recommended approach to have an `initialState` as the default parameter of a reducer. That way, it also acts as the data structure definition for the entire state.

File: src/ui/content-settings/__tests__/ContentSettingsCropRatiosContainer.test.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+const actionList = [
+  'onAdd',
+  'onDelete',
+  'onUpdate',
+];
+
+describe('ContentSettingsCropRatiosContainer', () => {
+  it('should render ContentSettingsCropRatios with correct props', () => {
+    shallow(
+      <ContentSettingsCropRatiosContainer store={store} />,
```
Comment (jeffgo10) [inline src/ui/content-settings/__tests__/ContentSettingsCropRatiosContainer.test.js]: Hi @raxkaynan. I tried running unit test and it seems there's an error in this area. Please check.

```
src/ui/content-settings/__tests__/ContentSettingsCropRatiosContainer.test.js
  ● ContentSettingsCropRatiosContainer › should render ContentSettingsCropRatios with correct props

    Invariant Violation: [React Intl] Could not find required `intl` object. <IntlProvider> needs to exist in the component ancestry.

      38 | describe('ContentSettingsCropRatiosContainer', () => {
      39 |   it('should render ContentSettingsCropRatios with correct props', () => {
    > 40 |     shallow(
         |     ^
      41 |       <ContentSettingsCropRatiosContainer store={store} />,
      42 |     ).dive().dive();
      43 | 

      at invariant (node_modules/invariant/invariant.js:40:15)
      at invariantIntlContext (node_modules/react-intl/lib/index.js:473:3)
      at new InjectIntl (node_modules/react-intl/lib/index.js:557:7)
      at ReactShallowRenderer.render (node_modules/react-test-renderer/cjs/react-test-renderer-shallow.development.js:723:26)
      at render (node_modules/enzyme-adapter-react-16/src/ReactSixteenAdapter.js:549:53)
      at fn (node_modules/enzyme-adapter-utils/src/Utils.js:99:18)
      at Object.render (node_modules/enzyme-adapter-react-16/src/ReactSixteenAdapter.js:549:18)
      at new ShallowWrapper (node_modules/enzyme/build/ShallowWrapper.js:265:22)
      at shallow (node_modules/enzyme/build/shallow.js:21:10)
      at Object.it (src/ui/content-settings/__tests__/ContentSettingsCropRatiosContainer.test.js:40:5)
```

using `testutils/helpers.js > mockRenderWithIntl` could eliminate this error.

**Comment by raxkaynan** [comment]
Done with the changes, @jeffgo10. Please review it again. Thanks!

**Pull request merged**